server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/task/DataTransferTaskContext.java

```diff
@@ -13,121 +13,68 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package com.oceanbase.odc.service.datatransfer.task;
 
-import java.util.Collections;
-import java.util.List;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
-import java.util.stream.Collectors;
-
-import org.apache.commons.lang.StringUtils;
 
-import com.oceanbase.odc.core.shared.Verify;
-import com.oceanbase.odc.service.flow.task.model.DataTransferTaskResult;
-import com.oceanbase.tools.loaddump.common.enums.ObjectType;
-import com.oceanbase.tools.loaddump.common.model.ObjectStatus;
-import com.oceanbase.tools.loaddump.common.model.Summary;
-import com.oceanbase.tools.loaddump.context.TaskContext;
+import com.oceanbase.odc.common.lang.Holder;
+import com.oceanbase.odc.plugin.task.api.datatransfer.DataTransferCallable;
+import com.oceanbase.odc.plugin.task.api.datatransfer.model.DataTransferTaskResult;
 
-import lombok.AccessLevel;
-import lombok.Getter;
-import lombok.NonNull;
-
-/**
- * {@link DataTransferTaskContext}
- *
- * @author yh263208
- * @date 2022-07-05 17:50
- * @since ODC_release_3.4.0
- */
-@Getter
 public class DataTransferTaskContext implements Future<DataTransferTaskResult> {
+    private final Future<DataTransferTaskResult> controlFuture;
+    private final Holder<DataTransferCallable> job;
 
-    @Getter(AccessLevel.NONE)
-    private final Future<DataTransferTaskResult> handle;
-    volatile TaskContext dataContext;
-    volatile TaskContext schemaContext;
+    public DataTransferTaskContext(Future<DataTransferTaskResult> controlFuture, Holder<DataTransferCallable> job) {
+        this.controlFuture = controlFuture;
+        this.job = job;
+    }
 
-    DataTransferTaskContext(@NonNull Future<DataTransferTaskResult> handle) {
-        this.handle = handle;
+    public DataTransferTaskResult getStatus() {
+        if (job.getValue() == null) {
+            return null;
+        }
+        return job.getValue().getStatus();
+    }
+
+    public double getProgress() {
+        if (job.getValue() == null) {
+            return 0.0;
+        }
+        return job.getValue().getProgress();
     }
 
     @Override
     public boolean cancel(boolean mayInterruptIfRunning) {
-        return handle.cancel(mayInterruptIfRunning);
+        if (job.getValue() != null) {
+            job.getValue().cancel(mayInterruptIfRunning);
```

Comment:
if `job.getValue().cancel(mayInterruptIfRunning);` returns a boolean, this method should return the and value between `job.getValue().cancel(mayInterruptIfRunning);` and `controlFuture.cancel(mayInterruptIfRunning);`

Reply:
done, added a bool return value on `DataTransferJob#cancel`.